File: backend/src/main/java/com/project/entities/MoodEntry.java
package com.project.entities;

import java.time.LocalDate;

import com.project.enums.Mood;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(
    name = "mood_entries",
    uniqueConstraints = {
        @UniqueConstraint(columnNames = {"user_id", "entry_date"})
    }
)
public class MoodEntry extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long moodId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Mood mood;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @Column(name = "entry_date", nullable = false)
    private LocalDate entryDate;
}

File: backend_dotnet/SafeMind.Core/Entities/MoodEntry.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using SafeMind.Core.Enums;

namespace SafeMind.Core.Entities
{
    [Table("mood_entries")]
    public class MoodEntry : BaseEntity
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [JsonPropertyName("moodId")]
        public long MoodId { get; set; }

        [Required]
        [ForeignKey(nameof(User))]
        [Column("user_id")]
        [JsonPropertyName("userId")]
        public long UserId { get; set; }

        [JsonPropertyName("user")]
        public User User { get; set; } = null!;

        [Required]
        [JsonPropertyName("mood")]
        public Mood Mood { get; set; }

        [Column(TypeName = "TEXT")]
        [JsonPropertyName("notes")]
        public string? Notes { get; set; }

        [Required]
        [Column("entry_date")]
        [JsonPropertyName("entryDate")]
        public DateOnly EntryDate { get; set; } // Use DateOnly for LocalDate equivalence in .NET 6+
    }
}
